quarkchain/cluster/shard_state.py

```diff
@@ -119,36 +120,48 @@ def init_from_root_block(self, root_block):
 
         self.header_tip = header_tip
         self.root_tip = root_block.header
+        header_tip_hash = header_tip.get_hash()
 
         self.db.recover_state(self.root_tip, self.header_tip)
         Logger.info(
             "[{}] Done recovery from db. shard tip {} {}, root tip {} {}".format(
                 self.full_shard_id,
                 self.header_tip.height,
-                self.header_tip.get_hash().hex(),
+                header_tip_hash.hex(),
                 self.root_tip.height,
                 self.root_tip.get_hash().hex(),
             )
         )
 
-        self.meta_tip = self.db.get_minor_block_meta_by_hash(self.header_tip.get_hash())
+        self.meta_tip = self.db.get_minor_block_meta_by_hash(header_tip_hash)
         self.confirmed_header_tip = confirmed_header_tip
-        self.evm_state = self.__create_evm_state()
-        self.evm_state.trie.root_hash = self.meta_tip.hash_evm_state_root
+        self.evm_state = self.__create_evm_state(
+            self.meta_tip.hash_evm_state_root, header_hash=header_tip_hash
+        )
         check(
-            self.db.get_minor_block_evm_root_hash_by_hash(self.header_tip.get_hash())
+            self.db.get_minor_block_evm_root_hash_by_hash(header_tip_hash)
             == self.meta_tip.hash_evm_state_root
         )
 
         self.__rewrite_block_index_to(
-            self.db.get_minor_block_by_hash(self.header_tip.get_hash()),
-            add_tx_back_to_queue=False,
+            self.db.get_minor_block_by_hash(header_tip_hash), add_tx_back_to_queue=False
         )
 
-    def __create_evm_state(self):
-        return EvmState(
+    def __create_evm_state(
+        self, root_hash: Optional[bytes], header_hash: Optional[bytes]
```

Comment:
I would suggest renaming root_hash to trie_root_hash to avoid confusion with root block hash

Reply:
or evm_state_root_hash